### PR TITLE
feat: Helm pull container registry validate/disable

### PR DESCRIPTION
**Pull request by shivani170**
Description
oci public changes in container registry

Fixes [#AB4425](https://dev.azure.com/DevtronLabs/Devtron/_boards/board/t/OSS%20adoption/Stories/?workitem=4425)

Type of change
 Bug fix (non-breaking change which fixes an issue)
 New feature (non-breaking change which adds functionality)
 Breaking change (fix or feature that would cause existing functionality to not work as expected)
 This change requires a documentation update
Checklist:
 The title of the PR states what changed and the related issues number (used for the release note).
 Does this PR require documentation updates?
 I've updated documentation as required by this PR.
 I have performed a self-review of my own code
 I have commented my code, particularly in hard-to-understand areas

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/dockerRegistry/Docker.tsx [hunk 1]
```diff
@@ -428,6 +434,10 @@ function DockerForm({
     const [errorValidation, setErrorValidation] = useState<boolean>(false)
     const [showHelmPull, setListRepositories] = useState<boolean>(false)
     const [isOCIRegistryHelmPush, setOCIRegistryHelmPush] = useState<boolean>(false)
+    const [validationError, setValidationError] = useState({ errtitle: '', errMessage: '' })
```
Comment (vivek-devtron) [inline src/components/dockerRegistry/Docker.tsx:437]: Please rename it to errTitle

File: src/components/dockerRegistry/Docker.tsx [hunk 2]
```diff
@@ -563,6 +572,36 @@ function DockerForm({
         toggleDefault(not)
     }
 
+    async function onClickValidate() {
+        setValidationStatus(VALIDATION_STATUS.LOADER)
+        // let isInvalid = isFormInvalid();
```
Comment (vivek-devtron) [inline src/components/dockerRegistry/Docker.tsx:577]: Please remove commented code

File: src/components/dockerRegistry/Docker.tsx [hunk 3]
```diff
@@ -563,6 +572,36 @@ function DockerForm({
...
+        // if (!isInvalid) {
+        //     toast.error("Some Required Fields are missing");
+        //     return
+        // }
+        let payload = getRegistryPayload(awsRegion)
+        payload.ociRegistryConfig = isPublic ? { CHART: OCIRegistryConfigConstants.PULL } : OCIRegistryStorageConfig
+
+        let promise = validateContainerConfiguration(payload)
+        await promise.then((response) => {
```
Comment (vivek-devtron) [inline src/components/dockerRegistry/Docker.tsx:586]: why are we adding await here?

File: src/components/dockerRegistry/Docker.tsx [hunk 4]
```diff
@@ -713,21 +752,21 @@ function DockerForm({
     }
 
     const handleOCIRegistryStorageAction = (key) => {
-        if (key === OCIRegistryConfigConstants.PUSH) {
```
Comment (vivek-devtron) [inline src/components/dockerRegistry/Docker.tsx:716]: Why did we remove the constants?

Comment (shivani170) [reply src/components/dockerRegistry/Docker.tsx:716]: Earlier it was unrelevant. Added relevant constant

**Pull request merged**